tools/cli/lib/api.c

```diff
@@ -671,6 +671,322 @@ TDNFCliRepoSyncCommand(
+             nCount += j;
+         }
+      }
+      if (nCount > 0)
+      {
+          dwError = TDNFAllocateMemory(nCount + 1, sizeof(char *), (void**)&ppszLines);
+          BAIL_ON_CLI_ERROR(dwError);
+          for (k = 0, i = 0; i < (int)dwCount; i++)
+          {
+              pPkgInfo = &pPkgInfos[i];
+
+              if (pPkgInfo->ppszDependencies)
+              {
+                  for (j = 0; pPkgInfo->ppszDependencies[j]; j++)
+                  {
+                      ppszLines[k++] = pPkgInfo->ppszDependencies[j];
+                  }
+              }
+              else if (pPkgInfo->ppszFileList)
+              {
+                  for (j = 0; pPkgInfo->ppszFileList[j]; j++)
+                  {
+                      ppszLines[k++] = pPkgInfo->ppszFileList[j];
+                  }
+              }
+          }
+
+          dwError = TDNFStringArraySort(ppszLines);
+          BAIL_ON_CLI_ERROR(dwError);
+
+          nSepSize = strlen("\n");
+          nSize = nSepSize * (nCount + 1);
+          for(i = 0; ppszLines[i]; i++)
+          {
+              if(i == 0 || strcmp(ppszLines[i], ppszLines[i-1]))
+                nSize += strlen(ppszLines[i]);
+          }
+          nSize++;
+
+          dwError = TDNFAllocateMemory(nSize, sizeof(char), (void**)&pszdependencies);
+          BAIL_ON_CLI_ERROR(dwError);
+
+          dwError = TDNFAllocateMemory(1, sizeof(char *), (void**)&ppPkgInfo->ppszDependencies);
+          BAIL_ON_CLI_ERROR(dwError);
+
+          for (j = 0; ppszLines[j]; j++)
+          {
+              if (j == 0)
+              {
+                   strcpy(&pszdependencies[p], ppszLines[j]);
+                   p += strlen(ppszLines[j]);
+              } else if (strcmp(ppszLines[j], ppszLines[j-1]))
+              {
+                   strcpy(&pszdependencies[p], "\n");
+                   p += nSepSize;
+                   strcpy(&pszdependencies[p], ppszLines[j]);
+                   p += strlen(ppszLines[j]);
+              }
+          }
+
+          dwError = TDNFAllocateString(pszdependencies, &ppPkgInfo->ppszDependencies[0]);
+          BAIL_ON_CLI_ERROR(dwError);
+    } else {
+          dwError = TDNFAllocateMemory(1, sizeof(char *), (void**)&ppPkgInfo->ppszDependencies);
+          BAIL_ON_CLI_ERROR(dwError);
+          ppPkgInfo->ppszDependencies[0] = strdup("(null)");
+    }
+
+cleanup:
+    if(pPkgInfos)
+    {
+        TDNFFreePackageInfoArray(pPkgInfos, dwCount);
+    }
+    TDNF_CLI_SAFE_FREE_MEMORY(ppszLines);
+    TDNF_CLI_SAFE_FREE_MEMORY(pszdependencies);
+    TDNFCliFreeRepoQueryArgs(pRepoqueryArgs);
+    return dwError;
+
+error:
+    goto cleanup;
+}
+
+uint32_t find_dependencies(PTDNF_CLI_CONTEXT pContext, int nquery_arg, char **query_argv, void *data, const int tdwCount, const int pdwCount, PTDNF_CMD_ARGS* ppCmdArgs)
+{
+    uint32_t dwError = 0;
+    PTDNF_CMD_ARGS pCmdArgs = NULL;
+
+    if(pdwCount == 0)
+    {
+       dwError = TDNFCliParseArgs(nquery_arg, query_argv, &pCmdArgs);
```

Comment:
We shouldn't create our own command line and reparse arguments, then re-run `TDNFRepoQurey()`. There needs to be another way.

Reply:
Removed custom command line parser and implemented the approach two for queryformat implementation.